### PR TITLE
[recipe] feat: Add Qwen3.5 LLM-only SFT recipes with skip_megatron_pa…

### PR DESCRIPTION
**Pull request by cuichenx**
## What does this PR do?

Adds `skip_megatron_param_globs` to `load_weights_hf_to_megatron` and `stream_weights_hf_to_megatron` on `MegatronModelBridge`, allowing selective parameter skipping during HF→Megatron weight loading via fnmatch patterns.

Uses this mechanism to ship 8 Qwen3.5 LLM-only SFT recipes (`qwen35_llm_{800m,2b,4b,9b,27b,35b_a3b,122b_a10b,397b_a17b}_sft_config`) that load language-model weights from Qwen3.5-VL checkpoints while skipping the vision tower (`*vision_model*`). Includes LoRA/PEFT variants for the dense sizes.

## Changelog

- `MegatronModelBridge.load_weights_hf_to_megatron`: add `skip_megatron_param_globs` kwarg
- `MegatronModelBridge.stream_weights_hf_to_megatron`: add `skip_megatron_param_globs` kwarg
- `AutoBridge.to_megatron_provider`: thread `skip_megatron_param_globs` through to the provider
- `megatron_param_matches_skip_globs`: new helper (fnmatch on global + local param name)
- `src/megatron/bridge/recipes/qwen_vl/qwen35_llm.py`: 8 LLM-only SFT + PEFT configs
- `tests/functional_tests/launch_scripts/h100/active/L1_Launch_models_qwen35_llm.sh`: CI launch script (L1 tier, runs T2.2/T2.5–T2.8)
- `tests/functional_tests/test_groups/models/qwen_vl/`: 5 functional test scripts (see below)

### Tier 1 — unit tests (no GPU)

| ID | File | What it checks | Result |
|---|---|---|---|
| T1.1–T1.3 | `test_megatron_param_skip_globs.py` | fnmatch logic, edge cases, `load_weights` path | ✅ PASS |
| T1.4 | `test_megatron_param_skip_globs.py` | `stream_weights` path respects skip globs | ✅ PASS |
| T1.5–T1.7 | `test_megatron_param_skip_globs.py` | recipe configs expose correct skip globs | ✅ PASS |

### Tier 2 — functional tests (GPU, real HF checkpoint)

| ID | File | What it checks | Result |
|---|---|---|---|
| T2.1 | — | VL training parity (main vs PR): iter-1 Δ=0.0, iter-5 Δ=7e-5 | ✅ PASS |
| T2.2 | `test_qwen35_llm_sft_skip_vision.py` | `stream_weights` yields no `*vision_model*` tasks; LM tasks present | ✅ PASS (3/3) |
| T2.3 | — | LLM-only 5-step smoke: losses finite and decreasing | ✅ PASS |
| T2.4 | — | Stripped-index test: N/A (0.8B is single-shard) | — |
| T2.5 | `test_qwen35_llm_optimizer_state.py` | Vision frozen (153 params), LM trainable (230 params), peak 17.3 GB | ✅ PASS |
| T2.6 | `test_qwen35_llm_weight_parity.py` | `embed_tokens` and `final_layernorm` weights bitwise-equal to HF source | ✅ PASS |
| T2.7 | `test_qwen35_llm_resume.py` | Save at step 3, resume starts at step 3 (not 0), all 5 losses finite | ✅ PASS |
| T2.8 | `test_qwen35_llm_tp_pp.py` | TP=2: losses [1.3856, 1.2917] | ✅ PASS |
| T2.8 | `test_qwen35_llm_tp_pp.py` | PP=2: losses [1.3808, 1.2870] (last pipeline stage) | ✅ PASS |

### CODE REVIEW

**Comment by cuichenx** [comment]
/ok to test https://github.com/NVIDIA-NeMo/Megatron-Bridge/pull/3037/commits/6d59e6483d2b2b528639df690fa92b01bb0c0c8a

File: src/megatron/bridge/recipes/qwen_vl/qwen35_llm.py [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+    max_lr: float,
+    min_lr: float,
+    train_iters: int = 1000,
+    global_batch_size: int = 128,
+    micro_batch_size: int = 1,
+) -> None:
+    """Apply Qwen3.5 LLM-only SFT settings on top of ``_sft_common``."""
+    seq_len = cfg.dataset.seq_length
+
+    cfg.model = AutoBridge.from_hf_pretrained(hf_path).to_megatron_provider(
```
Comment (yaoyu-33) [inline src/megatron/bridge/recipes/qwen_vl/qwen35_llm.py]: will megatron still initialize vision part of model?

**Comment by cuichenx** [comment]
will remove skip_megatron_param_globs

**Comment by cuichenx** [comment]
Addresses the open punch list from review:

- `2e37443b4` — drop `skip_megatron_param_globs` (orphan API: the recipe was refactored to use `add_encoder=False` and no other caller exists).
- `ae69676df` — rename Qwen3.5-VL provider field `init_vision_model` → `add_encoder` to match the sibling `Qwen3VLModelProvider`.
- `76a6831b7` — strengthen `test_qwen35_llm_resume.py`: perturb the first trainable parameter to a sentinel value at the last step of phase 1, then assert it round-trips at `on_train_start` of phase 2. Catches the failure mode where the HF pre-wrap hook silently shadows the checkpoint loader.
- `b7c21d19b` — extend `test_qwen35_llm_weight_parity.py` with L2-norm parity checks for layer-0 `linear_qkv.weight` (Q+K+V fusion) and `linear_fc1.weight` (gate+up fusion). L2 norm is layout-invariant, so it catches scaling drift / wrong-layer / random-init regressions without re-implementing the bridge's fusion layout in test code.
- `670891814` — add `PEAK_MEM_GB_CEILING=25.0` regression guard to `test_qwen35_llm_optimizer_state.py` (env-overridable). Reference measurement was ~17.3 GB; full Qwen3.5-VL would push past 30 GB.
- `b9ed11f99` — drop no-op `freeze_vision_model = True` / `freeze_vision_projection = True` from the recipe and stale assertions from the unit test. With `add_encoder=False` there is no vision module for the freeze hook to apply to.

All six commits signed-off; pre-commit (ruff + format) clean.

**Comment by cuichenx** [comment]
superseded by #3769
